tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -0,0 +1,198 @@
+assert = chai.assert
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+class FakeAnchor extends Annotator.Anchor
+
+  _getSegment: -> "html segment for " + @id
+
+  constructor: (manager, annotation, target) ->
+    super manager, annotation, target, 42, 42,
+      "fake quote for" + target.id
+
+    @id = "fake anchor for " + target.id
+
+describe 'Annotator.Plugin.EnhancedAnchoring', ->
+  sandbox = null
+
+  createAnchoringManager = ->
+    annotator =
+      publish: (event) ->
+
+    am = new Annotator.Plugin.EnhancedAnchoring()
+    am.annotator = annotator
+    am.pluginInit()
+
+    am._chooseAccessPolicy()
```

Comment:
What's the public method that triggers this call? I had a look and it's called by `_scan` but I couldn't see what calls scan.